docs/0002-N-tuple-tree.md

```diff
@@ -0,0 +1,180 @@
+* The characteristics of the underlying storage and associated code libraries. Although not the case in the past, modern storage systems can generally handle tens of thousands of files in a directory without difficulty. It is more likely that the code libraries and tools used to access and parse these systems will encounter some performance limitations when handling large numbers of files. In particular, Linux command-line wildcard expansions are typically limited to just under 128K characters which equates to around 4000 directory names if they have 32 characters. 
+* Human readability is also reduced for long lists of files, which may make recovery *in extremis* more difficult.
+
+For a tuple size of 3, our example UUID of "f81d4fae7dec11d0a76500a0c91e6bf6" would be split up into a path beginning "/f81/d4f/ae7/...".
+
+### number of tuples
+
+In practice, you may wish to limit the depth of the OCFL Storage Hierarchy tree to avoid overly deep nesting of directories. The **number of tuples** determines this depth. Splitting the entire identifier down into tuples may not be necessary since the number of objects to be stored is often much less than all the possible values for the source identifier. 
+
+For example, if we split the example UUID into size 3 tuples, each directory can contain 4096 subdirectories so, with the number of tuples also set to three the resulting tree would have 4096^3 (=68719476736) directories, each of which could contain one or more objects. All the objects with UUID's that begin f81d4fae7... would be stored in OCFL object roots in the directory /f81/d4f/ae7/. However, if the UUID's are reasonably pseudo-randomly distributed, the likelihood of many object identifiers sharing even the first 9 characters is quite low until a signficant number of objects have been created.
+
+### short object root
+
+Once a Storage Hierarchy has been created, the name of each OCFL Object Root directory should also be determined from the identifier in a consistent manner. The default approach is to use the full (stripped) identifier but, if identifiers are long or there is the need to keep Storage Hierarchy paths short because of object complexity, there is the option to just use the portion of the identifier that remains after Storage Hierarchy path generation. To continue the UUID example the full OCFL Object Root path could be:    
+* **short object root = FALSE** /f81/d4f/ae7/f81d4fae7dec11d0a76500a0c91e6bf6/
+* **short object root = TRUE** /f81/d4f/ae7/dec11d0a76500a0c91e6bf6/
+
+## Examples
+
+These examples are taken from the OCFL Implementation notes:
+
+* *Flat*: Each object is contained in a directory with a name that is simply derived from the unique identifier of the object.
+  * identifer length = 12
+  * case mapping = ToLower
+  * invert mapping = FALSE
+  * tuple size = 12
+  * number of tuples = 0
```

Comment:
Perhaps it would make more sense if both of these values were 0?

Reply:
I can see the sense in that.

Reply:
Changed to zero in PR and amended the text to make it clear what zero means